### PR TITLE
Added RecursivePage

### PR DESCRIPTION
**Pull request by NOTSAHLO**

### CODE REVIEW

File: react-app/src/components/UI/Recursive.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import styles from '../../styles/Recursive.module.css'
+
+export default function Recursive(props: {message: string, depth: number, maxDepth: number}) {
```
Comment (RZEPI) [inline react-app/src/components/UI/Recursive.tsx]: there is no use in using props as object, it's better to destructure it

File: react-app/src/pages/RecursivePage.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { useState } from 'react';
+import styles from "../styles/RecursivePage.module.css";
+import Recursive from "../components/UI/Recursive.tsx";
+export default function RecursivePage() {
+    const message = "This component was created recursively. It's depth is";
+    const depth = 1;
+    const [maxDepth, setMaxDepth] = useState(0);
+    const [inputValue, setInputValue] = useState(0);
+
+    const content = (
```
Comment (RZEPI) [inline react-app/src/pages/RecursivePage.tsx]: don't declare this variable, add jsx to the return

File: react-app/src/App.tsx [hunk 3]
```diff
@@ -12,6 +13,7 @@ const router = createBrowserRouter([
     children: [
       { index: true, element: <HomePage /> },
       { path: "table", element: <TablePage /> },
+      { path: "recursive", element: <RecursivePage /> },
```
Comment (RZEPI) [inline react-app/src/App.tsx]: inconsistent url with vue

**Review by RZEPI — CHANGES_REQUESTED** [review]

File: react-app/src/App.tsx (hunk 3 above)
Comment (NOTSAHLO) [reply react-app/src/App.tsx]: adjusted

File: react-app/src/pages/RecursivePage.tsx (hunk 2 above)
Comment (NOTSAHLO) [reply react-app/src/pages/RecursivePage.tsx]: change implemented

File: react-app/src/components/UI/Recursive.tsx (hunk 1 above)
Comment (NOTSAHLO) [reply react-app/src/components/UI/Recursive.tsx]: implemented

**Review by RZEPI — COMMENTED** [review]
now those elements don't diff from one another, what do you think about a font color saturation change  in child component

File: react-app/src/components/UI/RecursiveComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import styles from '../../styles/RecursiveComponent.module.css';
+
+export default function RecursiveComponent({ message, depth, maxDepth }: { message: string, depth: number, maxDepth: number }) {
```
Comment (RZEPI) [inline react-app/src/components/UI/RecursiveComponent.tsx]: On vue branch, you have type for this component, make it consistent

**Review by RZEPI — CHANGES_REQUESTED** [review]

File: react-app/src/components/UI/RecursiveComponent.tsx (hunk 4 above)
Comment (NOTSAHLO) [reply react-app/src/components/UI/RecursiveComponent.tsx]: fixed inconsistency

**Review by RZEPI — DISMISSED** [review]

File: react-app/src/components/UI/RecursiveComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { RecursiveProps } from "../../models/RecursiveProps.ts";
+import styles from '../../styles/RecursiveComponent.module.css';
+
+const getColor = (depth : number): string => {
+  const hue = 190 + (depth * 3);
+  const saturation = 96+ (depth * 3);
```
Comment (marziel2001) [inline react-app/src/components/UI/RecursiveComponent.tsx]: In fact this should be secured against overflow over 255 value. If this doesn't crash, it does not mean it's valid.

File: react-app/src/styles/RecursiveComponent.module.css [hunk 6]
```diff
@@ -0,0 +1,4 @@
+.recursive-component {
+    text-align: center;
+    font-size: 13px;
```
Comment (marziel2001) [inline react-app/src/styles/RecursiveComponent.module.css]: I think that font size should be always responsive
image

File: react-app/src/styles/RecursivePage.module.css [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    border-radius: 8px;
+    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
+}
+
+input[type="number"] {
+    padding: 0.5em;
+    font-size: 1em;
+    border: 1px solid #ccc;
+    border-radius: 4px;
+     width: 220px;
```
Comment (marziel2001) [inline react-app/src/styles/RecursivePage.module.css]: indentation

File: react-app/src/styles/RecursivePage.module.css [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    text-align: center;
+  }
+
+  button {
```
Comment (marziel2001) [inline react-app/src/styles/RecursivePage.module.css]: indentation

**Review by marziel2001 — COMMENTED** [review]
Sorry for being the pickiest person in the world but things have to be perfect 🥇 ;)

File: react-app/src/components/UI/RecursiveComponent.tsx (hunk 5 above)
Comment (marziel2001) [reply react-app/src/components/UI/RecursiveComponent.tsx:6]: or maybe you wish it to start from black when it overflows, then it's ok

Comment (NOTSAHLO) [reply react-app/src/components/UI/RecursiveComponent.tsx]: changed.

File: react-app/src/styles/RecursiveComponent.module.css (hunk 6 above)
Comment (NOTSAHLO) [reply react-app/src/styles/RecursiveComponent.module.css]: changed.

File: react-app/src/styles/RecursivePage.module.css (hunk 7 above)
Comment (NOTSAHLO) [reply react-app/src/styles/RecursivePage.module.css]: fixed

File: react-app/src/styles/RecursivePage.module.css (hunk 8 above)
Comment (NOTSAHLO) [reply react-app/src/styles/RecursivePage.module.css]: fixed

**Review by RZEPI — CHANGES_REQUESTED** [review]
in case of generating 10k components the page crashes
image

File: react-app/src/pages/RecursivePage.tsx [hunk 9]
```diff
@@ -20,7 +26,7 @@ export default function RecursivePage() {
                         placeholder="Amount of components"
                         min="0"
                     />
-                    <button onClick={() => setMaxDepth(inputValue)}>Generate Components</button>
+                    <button onClick={() => handleGenerateComponents()}>Generate Components</button>
```
Comment (RZEPI) [inline react-app/src/pages/RecursivePage.tsx]: This could be a pointer to handleGenerateComponents istead of lambda function

**Review by RZEPI — CHANGES_REQUESTED** [review]

File: react-app/src/pages/RecursivePage.tsx (hunk 9 above)
Comment (NOTSAHLO) [reply react-app/src/pages/RecursivePage.tsx]: fixed

**Review by RZEPI — DISMISSED** [review]

**Review by grczk — APPROVED** [review]

**Review by RZEPI — APPROVED** [review]

**Pull request merged**